### ISSUE TITLE
TypeError [ERR_WORKER_UNSUPPORTED_OPERATION] [ERR_WORKER_UNSUPPORTED_OPERATION]: process.chdir() is not supported in workers

### ISSUE THREAD

**Issue by gyerts** [angular/angular-cli#22438]
# 🐞 Bug report

- [ ] new
- [ ] build
- [ ] serve
- [x] test
- [ ] e2e
- [ ] generate
- [ ] add
- [ ] update
- [ ] lint
- [ ] extract-i18n
- [ ] run
- [ ] config
- [ ] help
- [ ] version
- [ ] doc

### Is this a regression?

Yes it has been working with version `"@angular-devkit/build-angular": "0.1102.2"`, after upgrade to the `"@angular-devkit/build-angular": "12.2.14"` tests not working any more.

### Description

When running tests `bazel test //browserclient:test`, I see this error in the logs
<pre><code>
TypeError [ERR_WORKER_UNSUPPORTED_OPERATION] [ERR_WORKER_UNSUPPORTED_OPERATION]: process.chdir() is not supported in workers
</code></pre>

## 🔬 Minimal Reproduction

clone test project [https://github.com/gyerts/angular_bazel_architect_fork](https://github.com/gyerts/angular_bazel_architect_fork)
`cd rules_nodejs/examples/angular_bazel_architect_fork/`

run `bazel test //browserclient:test`
everything should work fine

open `browserclient/package.json`
change version of `@angular-devkit/build-angular` from "0.1102.2" to "12.2.14"
run `yarn install` to update `yarn.lock` file
run `bazel test //browserclient:test` <- now it fails

## 🔥 Exception or Error

<pre><code>
TypeError [ERR_WORKER_UNSUPPORTED_OPERATION] [ERR_WORKER_UNSUPPORTED_OPERATION]: process.chdir() is not supported in workers
    at process.unavailableInWorker [as chdir] (internal/process/worker_thread_only.js:12:11)
    at Object.<anonymous> (browserclient/__chdir.js__:1:9)
    at Module._compile (internal/modules/cjs/loader.js:1072:14)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1101:10)
    at Module.load (internal/modules/cjs/loader.js:937:32)
    at Function.Module._load (internal/modules/cjs/loader.js:778:12)
    at Module.require (internal/modules/cjs/loader.js:961:19)
    at Module._preloadModules (internal/modules/cjs/loader.js:1231:12)
    at loadPreloadModules (internal/bootstrap/pre_execution.js:468:5)
    at MessagePort.<anonymous> (internal/main/worker_thread.js:151:5)
TypeError [ERR_WORKER_UNSUPPORTED_OPERATION] [ERR_WORKER_UNSUPPORTED_OPERATION]: process.chdir() is not supported in workers
    at process.unavailableInWorker [as chdir] (internal/process/worker_thread_only.js:12:11)
    at Object.<anonymous> (browserclient/__chdir.js__:1:9)
    at Module._compile (internal/modules/cjs/loader.js:1072:14)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1101:10)
    at Module.load (internal/modules/cjs/loader.js:937:32)
    at Function.Module._load (internal/modules/cjs/loader.js:778:12)
    at Module.require (internal/modules/cjs/loader.js:961:19)
    at Module._preloadModules (internal/modules/cjs/loader.js:1231:12)
    at loadPreloadModules (internal/bootstrap/pre_execution.js:468:5)
    at MessagePort.<anonymous> (internal/main/worker_thread.js:151:5)
</code></pre>

## 🌍 Your Environment

<pre><code>
Angular CLI: 12.2.14
Node: 14.17.3
Package Manager: yarn 1.22.10
OS: darwin x64

Angular: 12.2.15
... animations, common, compiler, compiler-cli, core, forms
... platform-browser, platform-browser-dynamic, router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1202.14
@angular-devkit/architect-cli   0.1102.2
@angular-devkit/build-angular   12.2.14
@angular-devkit/core            12.2.14
@angular-devkit/schematics      12.2.14
@angular/cli                    12.2.14
@bazel/bazelisk                 1.11.0
@bazel/buildifier               4.2.4
@bazel/ibazel                   v0.15.10
@schematics/angular             12.2.14
rxjs                            6.6.7
typescript                      4.3.5
</code></pre>

**Anything else relevant?**
This project based on [bazel-test-angular-architect project](https://github.com/bazelbuild/rules_nodejs/tree/stable/examples/angular_bazel_architect), the only difference between [bazel-test-angular-architect project](https://github.com/bazelbuild/rules_nodejs/tree/stable/examples/angular_bazel_architect) and [my-forked-example](https://github.com/gyerts/angular_bazel_architect_fork)
* frontend code was moved to the browserclient folder
* into each bazel rule `architect` and `architect_test` was added `chdir = package_name()` parameter
* angular was upgraded from 11 to 12

**Comment by cfife-btig** [comment]
This issue has been affecting/blocking me too on v13.

I am not an expert at all in the build-angular tool or the angular-cli, but I figure this issue is probably related to suddenly using workers when it previously wasn't. Is it possible this commit could be a culprit? 

https://github.com/angular/angular-cli/issues/22438

I basically did a quick search of all the changes in build-angular from 12 to 13, and this appears to be the only one that has anything to do with workers. 

I am trying to test/verify myself but it's a little tricky to reproduce since my stack is based in angular 13.

Again, I'm not an expert, but the reason I'm thinking this could be worker related and not related to the `chdir` command is because the `process.chdir` command I think is being called by bazel, not build-angular.

**Comment by cfife-btig** [comment]
Well, on the weird off-chance that this helps someone out,

I spent a long time trying to reproduce the issue mentioned above including creating a bazel workspace with an angular 12 project in the subdirectory and I couldn't reproduce the issue, it just worked. And then I tried a hello-world angular 13 project and I also could run `architect_test` without any issue.

I spent a long time on this and what finally got my angular 13 project to work with `architect_test` was simply switching my `.scss` to `.css`. After that, I stopped getting the error messages as shown above. Really weird and honestly not sure how the hell that's a problem.

I have a bazel project with something like this layout.

`/root_dir/angular_project`

and my bazel rule looks like below

architect_test(
    name = "test",
    args = [
        "frontend:test",
    ],
    chdir = package_name(),
    configuration_env_vars = ["NG_BUILD_CACHE"],
    data = [
        ":angular_test_code",
        "@npm//:node_modules",
    ],
)


**Tl;dr,** the comment I posted above was wrong, it turns out something with using sass was the issue. Very confusing and have no idea what the relationship is with the error message about `process.chdir not supported in workers`

**Comment by kormide** [comment]
Fyi this has been fixed: https://github.com/bazelbuild/rules_nodejs/pull/3322/commits/602362a7cfc6696f05c9c7a5368b9affeae694ba

**Comment by alan-agius4** [comment]
Thanks @kormide for chiming in. Closing as per above.
